### ISSUE TITLE
How do I make this basic test pass?

### ISSUE THREAD

**Issue by smac89** [PyCQA/pylint#7266]
## Description
I've written a checker plugin and (with quite a bit of effort) written a simple unit test for it. After struggling with the unittest to make it pass, I finally got it to a point where I can see the checker run and produce the message I wanted.

## The problem
This the output I'm getting:
```
Expected:
MessageTest(msg_id='dangerous-cascade', line=None, node=<Module.testapp.models l.0 at 0x7ff9866690d0>, args=('testapp.mymodel', 'testapp.mymodeluser'), confidence=Confidence(name='UNDEFINED', description='Warning without any associated confidence level.'), col_offset=None, end_line=None, end_col_offset=None)

Got:
MessageTest(msg_id='dangerous-cascade', line=13, node=<Module.testapp.models l.0 at 0x7ff98577f040>, args=('testapp.mymodel', 'testapp.mymodeluser'), confidence=Confidence(name='UNDEFINED', description='Warning without any associated confidence level.'), col_offset=4, end_line=13, end_col_offset=64)
```

I literally do not care about the other stuff in that message, and as far as I'm concerned this test has passed, however I have not been able to actually pass the test because I haven't found a way to just test that the message with the given ID is shown.

Is there some other way of doing this either using pylint's test utilities or something else? I just need this thing to pass.

Thank you all.

### Documentation for future user

https://github.com/oppia/oppia/wiki/Custom-Pylint-checks#pylint-testing-utilities
https://pylint.pycqa.org/en/latest/development_guide/how_tos/custom_checkers.html#testing-a-checker

Pylint Version:
```
pylint 2.14.5
astroid 2.11.7
Python 3.9.13 (main, Jul 28 2022, 10:35:47)
[GCC 11.2.0]
```

### Additional context

I used python's builtin unittest to write the test. This is a snippet of how the code for the test looks:

```py
def setUp(self):
    super().setUp()
    self.checker_test_object = pylint.testutils.CheckerTestCase()
    self.checker_test_object.CHECKER_CLASS = ...
    self.checker_test_object.setup_method()

def test_finds_(self):
    code = ...
    ...
    ast_module = astroid.parse(
        code, module_name="testapp.models", apply_transforms=False
    )
    with self.checker_test_object.assertAddsMessages(
        pylint.testutils.MessageTest(
            msg_id="dangerous-cascade",
            args=("testapp.mymodel", "testapp.mymodeluser"),
            node=ast_module,
        ),
        ignore_position=True,
    ):
       self.checker_test_object.walk(ast_module) # walks the ast tree, gathering data
       self.checker_test_object.checker._run() # this part is that signals the linter to do its checks
```

**Comment by Pierre-Sassoulas** [comment]
I think you have to specify the line / column if you're using a unittest. We're mostly using functional test ourselves as it's easier in almost all case. See 
https://github.com/PyCQA/pylint/blob/20a2f129e82349f8a216c288905dee748ba55423/tests/test_functional.py#L42

https://github.com/PyCQA/pylint/blob/20a2f129e82349f8a216c288905dee748ba55423/tests/functional/d/dangerous_default_value.py#L10

We definitely need to document it for plugin creator, and also probably move the automated update mechanism to testutil so it can be used easily without copy pasting everything in ``tests/test_functional.py``.

**Comment by DanielNoord** [comment]
This question has been answered!

**Comment by juanma-clavero** [comment]
I hope this doesn't reopen this old thread

you can use the MessageTest, you need to pass the fields from the node itself. something like 

```python
        pylint.testutils.MessageTest(
            msg_id="dangerous-cascade",
            args=("testapp.mymodel", "testapp.mymodeluser"),
            node=ast_module,
            line=ast_module.lineno,
            col_offset=ast_module.col_offset,
            end_line=ast_module.end_lineno,
            end_col_offset=ast_module.end_col_offset,
        ),
```

some of the attribute names may change, depending on the actual ast node you're using
